core/src/banking_stage/unified_scheduler.rs

```diff
@@ -0,0 +1,94 @@
+//! This module contains any integration code to put the unified scheduler subsystem into the
+//! banking stage, starting from `BankingPacketBatch` ingestion from the sig verify stage and to
+//! `Task` submission to the unified scheduler.
+//!
+//! These preprocessing for task creation can be multi-threaded trivially. At the same time, the
+//! maximum cpu core utilization needs to be constrained among this processing and the actual task
+//! handling of unified scheduler. Thus, it's desired to share a single thread pool for the two
+//! kinds of work. Towards that end, the integration was implemented as a callback-style, which is
+//! invoked (via `select!` on `banking_packet_receiver`) at each of unified scheduler handler
+//! threads.
+//!
+//! Aside from some limited abstraction leakage to make `select!` work at the
+//! solana-unified-scheduler-pool crate, almost all of these preprocessing are intentionally
+//! encapsulated into this module, at the cost of dynamic dispatch per each BankingPacketBatch to
+//! retain the unified scheduler agnostic scheduler over scheduling mode (block verification vs
+//! block production) as much as possible.
+//!
+//! Lastly, what the callback closure in this module does is roughly as follows:
+//! 1. Translate the raw packet bytes into some structs
+//! 2. Do various sanitization on them
+//! 3. Calculate priorities
+//! 4. Convert them to tasks with the help of provided BankingStageHelper (remember that pubkey
+//!    lookup for UsageQueue is also performed here; thus multi-threaded and off-loaded from the
+//!    scheduler thread)
+//! 5. Submit the tasks.
+
+#[cfg(feature = "dev-context-only-utils")]
+use qualifier_attr::qualifiers;
+use {
+    super::{
+        decision_maker::{BufferedPacketsDecision, DecisionMaker},
+        packet_deserializer::PacketDeserializer,
+        LikeClusterInfo,
+    },
+    crate::banking_trace::Channels,
+    agave_banking_stage_ingress_types::BankingPacketBatch,
+    solana_poh::poh_recorder::PohRecorder,
+    solana_runtime::{bank_forks::BankForks, root_bank_cache::RootBankCache},
+    solana_unified_scheduler_pool::{BankingStageHelper, DefaultSchedulerPool},
+    std::sync::{Arc, RwLock},
+};
+
+#[allow(dead_code)]
+#[cfg_attr(feature = "dev-context-only-utils", qualifiers(pub))]
+pub(crate) fn ensure_banking_stage_setup(
+    pool: &DefaultSchedulerPool,
+    bank_forks: &Arc<RwLock<BankForks>>,
+    channels: &Channels,
+    cluster_info: &impl LikeClusterInfo,
+    poh_recorder: &Arc<RwLock<PohRecorder>>,
+) {
+    let mut root_bank_cache = RootBankCache::new(bank_forks.clone());
+    let unified_receiver = channels.unified_receiver().clone();
+    let decision_maker = DecisionMaker::new(cluster_info.id(), poh_recorder.clone());
```

Comment:
I rugged your PR with #4724.

Decision maker caches the decision now internally for a short period (5ms) to avoid taking poh locks all the time when worker threads need them most.

I think you should be able to just make this `mut` and it'll work.